newIDE/app/src/Leaderboard/useLeaderboardReplacer.js

```diff
@@ -0,0 +1,415 @@
+// @flow
+import { Trans } from '@lingui/macro';
+import * as React from 'react';
+import Dialog, { DialogPrimaryButton } from '../UI/Dialog';
+import FlatButton from '../UI/FlatButton';
+import { Line } from '../UI/Grid';
+import Text from '../UI/Text';
+import LinearProgress from '@material-ui/core/LinearProgress';
+import {
+  Table,
+  TableBody,
+  TableHeader,
+  TableHeaderColumn,
+  TableRow,
+  TableRowColumn,
+} from '../UI/Table';
+import { ColumnStackLayout } from '../UI/Layout';
+import AuthenticatedUserContext, {
+  type AuthenticatedUser,
+} from '../Profile/AuthenticatedUserContext';
+import { duplicateLeaderboard } from '../Utils/GDevelopServices/Play';
+import { registerGame } from '../Utils/GDevelopServices/Game';
+import { toNewGdMapStringString } from '../Utils/MapStringString';
+
+const gd = global.gd;
+
+type ReplacePromptDialogProps = {|
+  leaderboardsToReplace: ?Array<string>,
+  onClose: () => void,
+  onTriggerReplace: () => void,
+|};
+
+export const ReplacePromptDialog = ({
+  leaderboardsToReplace,
+  onClose,
+  onTriggerReplace,
+}: ReplacePromptDialogProps) => {
+  const { authenticated, onLogin } = React.useContext(AuthenticatedUserContext);
+
+  return (
+    <Dialog
+      actions={
+        authenticated
+          ? [
+              <FlatButton
+                primary
+                key="register-game-later"
+                label={<Trans>Register later</Trans>}
+                onClick={onClose}
+              />,
+              <DialogPrimaryButton
+                primary
+                key="register-game-now"
+                label={<Trans>Register now</Trans>}
+                onClick={onTriggerReplace}
+              />,
+            ]
+          : [
+              <FlatButton
+                primary
+                key="login-later"
+                label={<Trans>Login later</Trans>}
+                onClick={onClose}
+              />,
+              <DialogPrimaryButton
+                label={<Trans>Login now</Trans>}
+                primary
+                onClick={() => onLogin()}
+                key="login-now"
+              />,
+            ]
+      }
+      title={<Trans>Leaderboards detected!</Trans>}
+      noMargin
+      open
+      maxWidth="sm"
+      onRequestClose={onClose}
+    >
+      <Line>
+        <ColumnStackLayout>
+          <Text>
+            <Trans>
+              We detected leaderboards that are not yet valid. Registering this
+              project with your GDevelop account would allow us to create them
+              in your stead so that your project is ready to use and publish
+              with minimal efforts.
```

Comment:
```suggestion
              This game is using leaderboards. GDevelop will create new leaderboards for this game in your account, so that the game is ready to be played and players can send their scores.
```

Reply:
(that's a suggestion)